jax/_src/lax/lax.py

```diff
@@ -6471,8 +6471,8 @@ def _const(example, val):
 _twos: Callable = partial(full_like, fill_value=2)
 _two: Callable = partial(full_like, shape=(), fill_value=2)
 
-dtype: Callable = dtypes.result_type
-_dtype: Callable = dtypes.result_type
+dtype: Callable = partial(dtypes.dtype, canonicalize=True)
```

Comment:
Maybe add a TODO to clean up these aliases.

Reply:
Would it be too much to do the cleanup here also? Perhaps we can introduce `dtypes.dtype_canonical` which is `partial(dtypes.dtype, canonicalize=True)` and if we need aliases elsewhere we can use `from dtypes import dtype, dtype_canonical`

I suspect that you considered this already and perhaps it leads to too many changes?

Reply:
The problem with importing `dtype` is that in many cases `dtype` is the name of a parameter in the function. So we'd have to import dtype as something other name... which is not so different than the current approach.

Reply:
Do we need both `dtype` and `_dtype`? Can we keep just `_dtype`?

Reply:
We could, except `lax.dtype` is an exported name, so it has to be defined somewhere... may as well be here.